models/mobilenet/mobilenet.hpp

```diff
@@ -0,0 +1,142 @@
+/**
+ * @file mobilenet.hpp
+ * @author Aakash Kaushik
+ * 
+ * Definition of MobileNet model.
+ * 
+ * For more information, kindly refer to the following paper.
+ *
+ * @code
+ * @article{Andrew G2017,
+ *  author = {Andrew G. Howard, Menglong Zhu, Bo Chen, Dmitry Kalenichenko,
+ *      Weijun Wang, Tobias Weyand, Marco Andreetto, Hartwig Adam},
+ *  title = {MobileNets: Efficient Convolutional Neural Networks for Mobile
+ *      Vision Applications},
+ *  year = {2017},
+ *  url = {https://arxiv.org/pdf/1704.04861}
+ * }
+ * @endcode
+ * 
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MODELS_MODELS_MOBILENET_MOBILENET_HPP
+#define MODELS_MODELS_MOBILENET_MOBILENET_HPP
+
+#define MLPACK_ENABLE_ANN_SERIALIZATION
+#include <mlpack.hpp>
+
+#include "separable_convolution.hpp"
+
+#include "./../../utils/utils.hpp"
+
+namespace mlpack {
+namespace models {
+
+/**
+ * Definition of a MobileNet CNN.
+ * 
+ * @tparam OutputLayerType The output layer type used to evaluate the network.
+ * @tparam InitializationRuleType Rule used to initialize the weight matrix.
+ */
+template<
+  typename MatType = arma::mat
+>
+class MobileNetType : public MultiLayer<MatType>
```

Comment:
I have a general question about the design.  I think it might be cleaner to simply have `MobileNetType` internally store an `FFN` that you add the layers to in the constructor.  I think several of the other models are built in this way.  Is there any particular advantage of this approach that I overlooked?  It seems like if you did just hold an `FFN` internally, you could use default copy/move constructors and operators and reduce the amount of code that you need.